server_admin/topics/identity-broker/social/bitbucket.adoc

```diff
@@ -0,0 +1,39 @@
+
+==== Bitbucket
+
+There are a number of steps you have to complete to be able to login to Bitbucket.  First, go to the `Identity Providers` left menu item
+and select `Bitbucket` from the `Add provider` drop down list.  This will bring you to the `Add identity provider` page.
+
+.Add Identity Provider
+image:{project_images}/bitbucket-add-identity-provider.png[]
+
+You can't click save yet, as you'll need to obtain a `Client ID` and `Client Secret` from Bitbucket.  One piece of data you'll need from this
+page is the `Redirect URI`.  You'll have to provide that to Bitbucket when you register {project_name} as a client there, so
+copy this URI to your clipboard.
+
+To enable login with Bitbucket you first have to register an application project in
+https://confluence.atlassian.com/bitbucket/oauth-on-bitbucket-cloud-238027431.html[OAuth on Bitbucket Cloud].
+
+NOTE: Bitbucket often changes the look and feel of application registration, so these directions might not always be up to date and the
+      configuration steps might be slightly different.
+
+
+.Add a New App
+image:images/bitbucket-developer-applications.png[]
+
+Click the `Add consumer` button.
+
+.Register App
+image:images/bitbucket-register-app.png[]
+
+You'll have to copy the `Redirect URI` from the {project_name} `Add Identity Provider` page and enter it into the
+`Authorization callback URL` field on the Bitbucket `Register a new OAuth application` page. Once you've completed this
+page you will be brought to the application's management page.
+
+NOTE: Please, make sure your app request permissions to read user's e-mail
+
+.Bitbucket App Page
+image:images/bitbucket-app-page.png[]
+
+You will need to obtain the client ID and secret from this page so you can enter them into the {project_name} `Add identity provider` page.
+Go back to {project_name} and specify those items.
```

Comment:
==== Bitbucket
There are a number of steps you have to complete to be able to login to Bitbucket.

First, open the `Identity Providers` left menu item and select `Bitbucket` from the `Add provider` drop down list. This will bring you to the `Add identity provider` page.

.Add Identity Provider
image:{project_images}/bitbucket-add-identity-provider.png[]

Before you can click `Save`, you must obtain a `Client ID` and `Client Secret` from Bitbucket.

NOTE: You will the `Redirect URI` from this page in a later step, which you will provide to Bitbucket when you register {project_name} as a client there.

.Add a New App
To enable login with Bitbucket you must first register an application project in
https://confluence.atlassian.com/bitbucket/oauth-on-bitbucket-cloud-238027431.html[OAuth on Bitbucket Cloud].

NOTE: Bitbucket often changes the look and feel of application registration, so what you see on the Bitbucket site may differ. If in doubt, see the Bitbucket documentation.

image:images/bitbucket-developer-applications.png[]

Click the `Add consumer` button.

.Register App
image:images/bitbucket-register-app.png[]

Copy the `Redirect URI` from the {project_name} `Add Identity Provider` page and enter it into the `Authorization callback URL` field on the Bitbucket `Register a new OAuth application` page.

On the same page, mark the `Email` and `Read` boxes under `Account` to allow your application to read user email.

.Bitbucket App Page
image:images/bitbucket-app-page.png[]

When you have completed this page you will be brought to the application's management page.

Reply:
The above is a suggested rewrite of the section.

Before you copy/paste, I have a question about the last sentence: they will be brought to which application's management page? We should specify. Another question: how do they complete the tasks on this page? Is there a save button?

Perhaps this would be better, replace that last sentence with "When you are done registering, click `Save`. This will open the application management page in Bitbucket."

Reply:
I just noticed that I accidentally moved the image header .Add a New App away from above image:images/bitbucket-developer-applications.png[] ... that was not an intentional part of the suggested rewrite.

Reply:
Also, this is needed at the very end, as shown:

When you are done registering, click Save. This will open the application management page in Bitbucket. Find the client ID and secret from this page so you can enter them into the {project_name} `Add identity provider` page.

To finish, return to {project_name} and enter them. Click `Save`.

Reply:
> Before you copy/paste, I have a question about the last sentence: they will be brought to which application's management page? We should specify. 

This is the application management page from Bitbucket:

![screenshot from 2018-03-28 09-23-47](https://user-images.githubusercontent.com/21758/38028748-334a2950-326a-11e8-8bc4-b98419f54fff.png)

I don't mind to be more specific, but it's pretty much part of the flow. I just provided similar steps based on GitHub's setup instructions.

> Another question: how do they complete the tasks on this page? Is there a save button?

They hit 'Save'. But assuming that we do not specify this for other integrations, I just omitted it.

> 
> Perhaps this would be better, replace that last sentence with "When you are done registering, click Save. This will open the application management page in Bitbucket."

I don't mind to change it. But it feels pretty obvious that people should hit 'Save', plus, that would imply that the instructions for other providers should be changed too. Don't you think?

Just let me know, whatever you think is great for our docs, I'm in.